web/app/js/components/ResourceDetail.jsx

```diff
@@ -53,67 +48,146 @@ export class ResourceDetailBase extends React.Component {
     return {
       namespace: resource.namespace,
       resourceName: resource.name,
-      resourceType: resource.type
+      resourceType: resource.type,
+      resource,
+      pollingInterval: 2000,
+      resourceMetrics: [],
+      podMetrics: [], // metrics for all pods whose owner is this resource
+      pendingRequests: false,
+      loaded: false,
+      error: null
     };
   }
 
-  banner = () => {
-    const {error} = this.props;
+  componentDidMount() {
+    this.loadFromServer();
+    this.timerId = window.setInterval(this.loadFromServer, this.state.pollingInterval);
+  }
 
-    if (!error) {
+  componentWillReceiveProps(newProps) {
+    // React won't unmount this component when switching resource pages so we need to clear state
+    this.api.cancelCurrentRequests();
+    this.setState(this.getInitialState(newProps.match, newProps.pathPrefix));
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+    this.api.cancelCurrentRequests();
+  }
+
+  loadFromServer() {
+    if (this.state.pendingRequests) {
+      return; // don't make more requests if the ones we sent haven't completed
+    }
+    this.setState({ pendingRequests: true });
+
+    let { resource } = this.state;
+
+    this.api.setCurrentRequests([
+      // inbound stats for this resource
+      this.api.fetchMetrics(
+        `${this.api.urlsForResource(resource.type, resource.namespace)}&resource_name=${resource.name}`
+      ),
+      // list of all pods in this namespace (hack since we can't currently query for all pods in a resource)
+      this.api.fetchPods(resource.namespace),
+      // metrics for all pods in this namespace (hack, continued)
+      this.api.fetchMetrics(
+        `${this.api.urlsForResource("pod", resource.namespace)}`
+      ),
+    ]);
+
+    Promise.all(this.api.getCurrentPromises())
+      .then(([resourceRsp, podListRsp, podRsp]) => {
+        let resourceMetrics = processSingleResourceRollup(resourceRsp);
+        let podMetrics = processSingleResourceRollup(podRsp);
+
+        // INEFFICIENT: get metrics for all the pods belonging to this resource.
+        // Do this by querying for metrics for all pods in this namespace and then filtering
+        // out those pods whose owner is not this resource
+        // TODO: fix (#1467)
+        let podBelongsToResource = _.reduce(podListRsp.pods, (mem, pod) => {
+          if (_.get(pod, resource.type) === resource.namespace + "/" + resource.name) {
```

Comment:
This unfortunately doesn't work for replication controllers because `resource.type` is "replicationcontroller", but the key in the ListPods JSON is "replicationController". I'm not sure what the best fix is.

Reply:
euuugh. I mean I can do some downcasing. I just wanted this to be a quick method, but we could also make this a comprehensive method that takes into account plurals too!

Reply:
omg I thought this was reference the shortname methods but it's not. maybe we'll just special case replication controllers here. it's yucky but I'm hoping this hack doesn't stay here long.